src/lib/utils/prepareTemplateProps.ts

```diff
@@ -1,6 +1,28 @@
 import uniq from './uniq';
+import { Template } from '../../types';
 
-function prepareTemplates(defaultTemplates = {}, templates = {}) {
+type TemplatesConfig = object;
+
+type Templates = {
+  [key: string]: Template;
```

Comment:
We don't leverage the generics here but not sure we need them.

Reply:
Yes, as far as I can tell we don't know anything about the type in this place, so I left it like this. Future usage (unlikely) can precise this further